File: LQDCoder/HELLO2/HELLO2.cpp
#include <iostream>
#include <string.h>

using namespace std;

string st1,st2;
int i,t,j,n;

int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    getline(cin,st1);
    t=stoi(st1);
    getline(cin,st2);
    n=st2.length();
    while (t != 0)
    {
        getline(cin,st1);
        j=0;
        for (i=0;i<st1.length();i++)
            if (st1[i] == st2[j]) j++;
        if (j == n) cout << "YES" << endl;
        else cout << "NO" << endl;
        t--;
    }
}

File: LQDCoder/HELLO2/HELLO2.pas
uses    crt;
var     st1,st2:ansistring;
        i,j,n:longint;
        kt:boolean;
begin
    clrscr;
    readln(n);
    readln(st2);
    while n <> 0 do 
        begin
            readln(st1);
            j:=1;
            kt:=false;
            for i := 1 to length(St1) do 
                begin
                    if (st1[i] = st2[j]) then inc(j);
                    if (j > length(st2)) then   
                        begin
                            kt:=true;
                            break;
                        end;
                end;
            if (kt = false) then writeln('NO')
            else writeln('YES');
            dec(n);
        end;
end.